### ISSUE TITLE
check the existence of a path

### ISSUE THREAD

**Issue by P-A-C** [OWASP/rbac#113]
I would like to know, if I create a path on permissions and after I do a check on this path with a false child permissions that does not exist, will it return me an error or will it return to me the parent role (that I have)?
example: 
"office / desk / computer" the computer does not exist. 
If I check "office / desk / computer "will return true if I have 'office/desk' ?

**Comment by abiusx** [comment]
As far as I remember it will return false. It should be easy to test though.
A non-existent child is not a child of anyone.

Paths are simply translated to hierarchies before being tested.

**Comment by P-A-C** [comment]
yes, but wouldn't it be interesting to have this kind of possibility?
If you check a path and the last argument of this path doesn't exist, but the parent permission does
that the exception is not thrown.
It may only interest me, but I would like to share with you.

**Comment by abiusx** [comment]
It shouldn't be hard to implement something like that.
Instead of checking from child to root, you can check root to child.

-A

**Comment by P-A-C** [comment]
I totally agree.
I'll try to do that.

**Comment by abiusx** [comment]
As far as I remember, currently the path to ID conversion is done using some
magical optimized MySQL query, which uses group_concat to make only one query to find
the parent and all the children. That's why it needs to be exact.

If you change that into a series of queries that retrieve the hierarchy, you can easily add
leniency. You can also try to work this into that magic query somehow!

**Comment by P-A-C** [comment]
Thank you
